### ISSUE TITLE
Incorrect comparison of nDPI_flow_info structures in ndpiSimpleIntegration

### ISSUE THREAD

**Issue by ol-andreyizrailev** [ntop/nDPI#1386]
https://github.com/ntop/nDPI/blob/8e512ff50331e2750d13b3d983f3b3216134ad18/example/ndpiSimpleIntegration.c#L362-L402
I believe the logic in `ip_tuples_compare()` is incorrect.
First of all, on the first line of the function (which is 365) `B->l3_type` should obviously be checked to be `L3_IP`, not `L3_IP6`.  The same problem presents in `ip_tuples_equal()`. 
Another issue here is the comparion logic itself. Let's take a look at the 366 to 375. You simplified code looks like this:
```
if (A->src < B->src || A->dst < B->dst)
{
    return -1;
}
if (A->src > B->src || A->dst > B->dst)
{
    return 1;
}
```
Let's now imagine that we pass to the function two structure pointers X and Y which have the following values: `X->src = 1, X->dst = 2, Y->src = 3, Y->dst = 1`.
If X is the first argument of the function and Y is the second one, the function will return -1 since X->src is less than Y->src.
But if Y is the first argument and X is the second one, the function will also return -1 since Y->src is greater than X->src and Y->dst is less than X->dst.
The first case implies that X is less than Y and the second case implies that Y is less than X. They are contradictory. The result of comparison must not be affected by the order of arguments passed to the function.
The same problem exists further on with IPv6 case and ports.
I suggest you do something like this:
```
if (A->src < B->src)
    return -1;
else if (A->src > B->src)
    return 1;

if (A->dst < B->dst)
    return -1;
else if (A->dst > B->dst)
    return 1;
```

This will also allow you to get rid of `ip_tuples_equal()` call in `ndpi_workflow_node_cmp()`.

**Comment by utoni** [comment]
I agree, these are two valid bugs. The first one was probably a typ0, but the second one is a logic fail.
Will fix that ASAP.
Thanks for reporting those!

**Comment by utoni** [comment]
Fixed with #1418.
